scripts/repl.js

```diff
@@ -57,6 +62,86 @@
   };
 
   /*
+   * Utils for working with parse.com for persistence
+   */
+  function ParseUtils () {}
+
+  var __parseLoadingCallbacks = null;
+  ParseUtils.prototype.load = function () {
+    if (__parseLoadingCallbacks) return;
+    __parseLoadingCallbacks = [];
+
+    var url = '//www.parsecdn.com/js/parse-1.3.5.min.js';
+    var script = document.createElement('script');
+    script.src = url;
+
+    hackToAllowCommonVarWhileLoadingParse = true;
+    script.addEventListener('load', function () {
+      Parse.initialize("ao4bnVLH8mmU1VUxDRFMzDDljabE0zJKhKJGQkZi", "1GnvnA6s9YgHVMUCyyQDxwJLkcnhRjhXvgnY6DGV");
+      hackToAllowCommonVarWhileLoadingParse = false;
+
+      __parseLoadingCallbacks.forEach(function (cb) {
+        cb();
+      });
+
+      __parseLoadingCallbacks = 'DONE';
+    });
+    document.head.appendChild(script);
+  };
+
+  ParseUtils.prototype.addLoadCallback = function (cb) {
+    __parseLoadingCallbacks.push(cb);
+  };
+
+  ParseUtils.prototype.afterLoad = function (cb) {
+    var self = this;
+
+    var callCb = function(){
+      var ReplItem = Parse.Object.extend("ReplItem");
+      cb.call(self, ReplItem);
+    };
+
+    if (__parseLoadingCallbacks === 'DONE') {
+      setTimeout(callCb, 0);
+    }
+    else {
+      self.load();
+      self.addLoadCallback(callCb);
+    }
+  };
+
+  ParseUtils.prototype.save = function (data, cb) {
+    this.afterLoad(function(ReplItem){
+      var obj = new ReplItem();
+
+      obj.save(data).then(function (obj) {
+        cb(null, obj.id);
+      }, function (err) {
+        cb(err);
+      });
+    });
+  };
+
+  ParseUtils.prototype.get = function (id, cb) {
+    this.afterLoad(function(ReplItem){
+      var query = new Parse.Query(ReplItem);
+      query.get(id)
+        .then(function (data) {
+        // give us a plain object without anything parse specific, making
+        // .get's output the same as .save's input
+        var cleanData = data.toJSON();
+        delete cleanData.objectId;
+        delete cleanData.createdAt;
+        delete cleanData.updatedAt;
```

Comment:
I believe it'll end up in state, which will end up in the URL.  I'll confirm it, though.